platform-codingcontest-sync/src/main/java/uno/cod/platform/server/codingcontest/sync/service/CodingcontestSyncService.java

```diff
@@ -1,11 +1,68 @@
 package uno.cod.platform.server.codingcontest.sync.service;
 
 
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uno.cod.platform.server.codingcontest.sync.dto.CodingcontestDto;
 import uno.cod.platform.server.codingcontest.sync.dto.ParticipationDto;
+import uno.cod.platform.server.core.domain.Challenge;
+import uno.cod.platform.server.core.domain.User;
+import uno.cod.platform.server.core.repository.ChallengeRepository;
+import uno.cod.platform.server.core.repository.ChallengeTemplateRepository;
+import uno.cod.platform.server.core.repository.UserRepository;
 
+import javax.transaction.Transactional;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+
+@Service
+@Transactional
 public class CodingcontestSyncService {
+    private final UserRepository userRepository;
+    private final ChallengeRepository challengeRepository;
+    private final ChallengeTemplateRepository challengeTemplateRepository;
+
+    @Autowired
+    public CodingcontestSyncService(UserRepository userRepository, ChallengeRepository challengeRepository, ChallengeTemplateRepository challengeTemplateRepository) {
+        this.userRepository = userRepository;
+        this.challengeRepository = challengeRepository;
+        this.challengeTemplateRepository = challengeTemplateRepository;
+    }
 
-    public void updateUserFromCodingcontest(ParticipationDto dto) {
+    public User updateUserFromCodingcontest(ParticipationDto dto) {
+        User user = userRepository.findByUsername(dto.getName());
+        if(user==null){
+            user = mapParticipationDto(dto);
+        }else{
+            user.setPassword(dto.getPassword());
+        }
+        return userRepository.save(user);
+    }
+
+    public void createContest(CodingcontestDto dto) {
+        String canonicalName = "ccc-" + dto.getLocation() + "-" + dto.getGameName();
```

Comment:
i'd use the received uuid for the canonicalName, the game switches too often afterwards ^^